src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncSingleLocalizationIT.java

```diff
@@ -0,0 +1,297 @@
+                ProductData::getMetaDescription, SetMetaDescription::of);
+        }
+        if (updateAction instanceof SetMetaKeywords) {
+            return filterLocalizedField(newDraft, oldProduct, locale, ProductDraft::getMetaKeywords,
+                ProductData::getMetaKeywords, SetMetaKeywords::of);
+        }
+        return Optional.of(updateAction);
+    }
+
+    /**
+     * Checks if the localized field value of the supplied {@link Locale} is different between the old and the new
+     * resource, if it is different, then an update action is generated with that change only of this localized field.
+     * if the values are not different, then an empty optional is returned.
+     *
+     * @param newDraft                the new product draft.
+     * @param oldProduct              the old existing product.
+     * @param locale                  the locale of the localized field to sync.
+     * @param newLocalizedFieldMapper mapper function to access the localized field on the new product draft.
+     * @param oldLocalizedFieldMapper mapper function to access the localized field on the old existing product.
+     * @param updateActionMapper      mapper function to build the update action to sync the localized field of the
+     *                                product.
+     * @return an optional containing an update action if the localized field with the specific locale has changed or
+     *         empty otherwise.
+     */
+    private static Optional<UpdateAction<Product>> filterLocalizedField(@Nonnull final ProductDraft newDraft,
+                                                                        @Nonnull final Product oldProduct,
+                                                                        @Nonnull final Locale locale,
+                                                                        @Nonnull final Function<ProductDraft,
+                                                                            LocalizedString> newLocalizedFieldMapper,
+                                                                        @Nonnull final Function<ProductData,
+                                                                            LocalizedString> oldLocalizedFieldMapper,
+                                                                        @Nonnull final Function<LocalizedString,
+                                                                            UpdateAction<Product>> updateActionMapper) {
+        final String newLocaleValue = newLocalizedFieldMapper.apply(newDraft).get(locale);
+        final LocalizedString oldLocalizedField = oldLocalizedFieldMapper.apply(oldProduct.getMasterData().getStaged());
+        final String oldLocaleValue = oldLocalizedField.get(locale);
+        if (!Objects.equals(newLocaleValue, oldLocaleValue)) {
+            LocalizedString withLocaleChange = oldLocalizedField;
+            if (oldLocaleValue != null) {
```

Comment:
If oldLocaleValue is null than it should ensure that newLocalizedField has French locale only, right?

Reply:
if oldLocalValue is null, then it will set the new French locale (if it exists). But it shouldn't touch other locales.

Reply:
The thing is that for some properties (like product description) "oldLocalizedField" might be not set yet so you will run into NPE.

Reply:
Ah, now I know what you mean. You are right: I fixed the logic here: 2f3bb99